src/vm/threads.h

```diff
@@ -1903,6 +1904,30 @@ class Thread: public IUnknown
         return m_pFrame;
     }
 
+    //--------------------------------------------------------------
+    // Returns innermost active GCFrame.
+    //--------------------------------------------------------------
+    PTR_GCFrame GetGCFrame()
+    {
+        SUPPORTS_DAC;
+
+#ifndef DACCESS_COMPILE
+#ifdef _DEBUG_IMPL
+        WRAPPER_NO_CONTRACT;
+        if (this == GetThreadNULLOk())
+        {
+            void* curSP;
+            curSP = (void *)GetCurrentSP();
+            _ASSERTE((m_pGCFrame == NULL) || (curSP <= m_pGCFrame && m_pGCFrame < m_CacheStackBase));
+        }
+#else
+        LIMITED_METHOD_CONTRACT;
```

Comment:
This is unreachable code. `_DEBUG_IMPL` is equivalent to `_DEBUG && !DACCESS_COMPILE`. This can be just:

```
#ifdef _DEBUG_IMPL
        WRAPPER_NO_CONTRACT;
        if (this == GetThreadNULLOk())
        {
            void* curSP;
            curSP = (void *)GetCurrentSP();
            _ASSERTE((m_pGCFrame == NULL) || (curSP <= m_pGCFrame && m_pGCFrame < m_CacheStackBase));
        }
#endif
        return m_pGCFrame;
```

Reply:
I see. The same is true for the GetFrame above then.